repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala

```diff
@@ -82,12 +89,15 @@ class UriStripperFilter @Inject()(configurationService: ConfigurationService)
         }
 
         token = Some(uriTokens.remove(config.getTokenIndex))
-        wrappedRequest.setRequestURI(joinTokensIntoUri(uriTokens))
+        wrappedRequest.setRequestURI(joinTokensIntoPath(uriTokens))
       }
 
       filterChain.doFilter(wrappedRequest, wrappedResponse)
 
+      wrappedResponse.uncommit()
+
       val originalLocation = wrappedResponse.getHeader(CommonHttpHeader.LOCATION.toString)
+      // todo: What if the path is "/<tenant-id>"? The location header wouldn't be modified. Is that the correct behavior?
```

Comment:
Original dev is Meynard:
https://github.com/rackerlabs/repose/blob/e48463d4a20a776d728c2cf27c50d2068bfbd9bd/project-set/components/uri-stripper/src/main/java/com/rackspace/papi/components/uritranslation/UriStripperHandler.java